### ISSUE TITLE
Problem with ROLLUP and sumMap

### ISSUE THREAD

**Issue by wiesenfeK** [ClickHouse/ClickHouse#32935]
**Describe what's wrong**

When using sumMap along with the `WITH ROLLUP` modifier, results seem wrong.

**Does it reproduce on recent release?**

I have the bug on v21.12.2.17

**How to reproduce**

* Clickhouse v21.12.2.17

```
CREATE TABLE bar 
(
    id String,
    n UInt32,
    w Map(UInt32, UInt32),
    y UInt32
)   
ENGINE = MergeTree()
PRIMARY KEY(id);
```

* Queries to run that lead to unexpected result

```
INSERT INTO bar 
VALUES('0',1, {1:1},1 ),('2',2, {1:2},2 ), ('3',1, {1:1},1 ), ('4',2, {1:2},2 );

SELECT 	sum(y), sumMap( w ), n
FROM bar
GROUP BY n
WITH ROLLUP 
ORDER BY n ASC;
```

**Expected behavior**

The expected result would be:
```
sum(y)|sumMap(w)|n|
------|---------|-|
     6|{1=6}    |0|
     2|{1=2}    |1|
     4|{1=4}    |2|
```

And this is what I obtain: 
(The difference is in the sumMap of the last row).

``` 
sum(y)|sumMap(w)|n|
------|---------|-|
     6|{1=6}    |0|
     2|{1=2}    |1|
     4|{1=6}    |2| <--- error on this row, in the sumMap(w) column. Expecting: {1=4}
```

Note that without ROLLUP, the aggregation is fine.

```
SELECT 	sum(y), sumMap( w ), n
FROM bar
GROUP BY n
ORDER BY n ASC;
```

Leads to:

``` 
sum(y)|sumMap(w)|n|
------|---------|-|
     2|{1=2}    |1|
     4|{1=4}    |2|
```

**Comment by wiesenfeK** [comment]
It happens with the `WITH CUBE` modifier as well.

**Comment by Algunenano** [comment]
This was fixed at some point in 22.3 and I see we have similar tests now
